src/core/type/type.ts

```diff
@@ -93,7 +96,7 @@ export interface IType<C, S, T> {
    *
    * @returns An instance of that type.
    */
-  create(snapshot?: C, env?: any): this["Type"]
+  create(snapshot?: C | ExcludeReadonly<T>, env?: any): this["Type"]
```

Comment:
This allows this test to continue passing:

https://github.com/mobxjs/mobx-state-tree/blob/c2f459b6126d4d1a990837541ced7f81f1490033/__tests__/core/type-system.test.ts#L1249-L1250

In other words, we want the type checker to let users know when they pass in a value for a getter view. I'm not sure whether or not this is the right decision, but this change is an addition (i.e., is backwards compatibile) to existing type checking, so I say we go for it and relax it if users encounter issues with it.